### PR TITLE
Progress indicator

### PR DESCRIPTION
**Pull request by alecmarcus**
# What this does
- visualizes progress of an entry (based on scroll) in the navbar
- adds a random dingbat at the end of every entry
- fixes a bug in Print.js where `data-fill-mode` was not being respected when the animation stopped due to idle

# What's included
- new class, `ProgressBar`
- new class, `RandomDingbat`

### CODE REVIEW

**Comment by alecmarcus** [comment]
Progress bar width is off, stops too early and reaches about `125vw`

**Review by stephalee — APPROVED** [review]
💯 💯 💯

**Review by bchiang7 — APPROVED** [review]
LGTM!

File: src/assets/scripts/ProgressBar.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { throttle } from './utils';
+
+class ProgressBar {
+  constructor() {
+    this.entryContentEl = document.getElementById('js-entry-content');
+    this.navigationEl = document.getElementById('js-entry-nav');
+    this.progressIndicator = document.getElementById('js-progress-indicator');
+    this.percentCompleteTextWrapper = document.getElementById('js-percent-complete');
+
+    this.animateThrottled = () => throttle(this.animate(), 16.66667); // 60fps
```
Comment (bchiang7) [inline src/assets/scripts/ProgressBar.js:10]: is this used anywhere?

Comment (alecmarcus) [reply src/assets/scripts/ProgressBar.js:10]: nope!

**Pull request merged**